src/lib/hooks/use-best-quote/utils/issuance/index.ts

```diff
@@ -34,30 +48,74 @@ export async function getEnhancedRedemptionQuote(
         indexTokenAmount
       )
     console.log('componentsUnits:', componentsUnits)
+    // TODO:
+    const outputTokenAmount = BigInt(0)
+
+    const sender = await signer.getAddress()
+
+    const callData = encodeFunctionData({
+      abi: DebtIssuanceModuleV2Abi,
+      functionName: 'redeem',
+      args: [
+        Ethereum2xFlexibleLeverageIndex.address! as Address,
+        indexTokenAmount,
+        sender as Address,
+      ],
+    })
+
+    const transaction: QuoteTransaction = {
+      chainId: 1,
+      from: sender,
+      to: contract,
+      data: callData,
+      value: undefined,
+    }
+
+    const defaultGas = getFlashMintGasDefault(inputToken.symbol)
+    const defaultGasEstimate = BigNumber.from(defaultGas)
+    console.log('gas', defaultGas, defaultGasEstimate.toString())
+    const gasEstimatooor = new GasEstimatooor(signer, defaultGasEstimate)
+    const canFail = false
+    const gasEstimate = await gasEstimatooor.estimate(transaction, canFail)
+    const gasCosts = gasEstimate.mul(gasPrice)
+    const gasCostsInUsd = getGasCostsInUsd(gasCosts, request.nativeTokenPrice)
+    transaction.gasLimit = gasEstimate
+    console.log('gasLimit', transaction.gasLimit.toString())
+
+    const inputTokenAmountUsd =
+      parseFloat(formatUnits(indexTokenAmount, inputToken.decimals)) *
+      inputTokenPrice
+    const outputTokenAmountUsd =
+      parseFloat(formatUnits(indexTokenAmount, inputToken.decimals)) *
+      inputTokenPrice
```

Comment:
Most of the above works similar to the other quote functions. Check `flashmint` or `0x` for reference.